micromanager_gui/main_window.py

```diff
@@ -147,14 +149,156 @@ def add_frame_explorer(self, name, array):
             self.viewer.add_image(array, name=layer_name)
 
     # TO DO: add the file name form the save box
-    def _on_mda_frame(self, img, event):
-        print("got frame", img, event)
-        name = "mda"
+    def _on_mda_frame(self, image, event):
+
+        sequence = event.sequence
+
+        event.index.setdefault('t', 0)
+        event.index.setdefault('z', 0)
+
+        t_stack_length = len(sequence.time_plan) or 1
+        p_stack_length = len(sequence.stage_positions)
+        z_stack_length = len(sequence.z_plan) or 1
+        c_stack_length = len(sequence.channels)
+
+        event_index_t = event.index["t"]
+        event_index_p = event.index["p"]
+        event_index_z = event.index["z"]
+        event_index_c = event.index["c"]
+        
+        file_name = 'mda'
+
+        layer_name = f'{file_name}_[{sequence.axis_order}]_p{p_stack_length}_t{t_stack_length}_z{z_stack_length}_c{c_stack_length}'
```

Comment:
```suggestion
        layer_name = (
            f'{file_name}_[{sequence.axis_order}]_p{p_stack_length}_'
            f't{t_stack_length}_z{z_stack_length}_c{c_stack_length}'
        )
```

(later we can add auto-formatting, but try to keep lines less than ~80 chars)